src/lib/reducer.js

```diff
@@ -1,287 +1,274 @@
-import convert from './convert';
-
-function reduce(node, precision) {
-  if (node.type === "MathExpression") 
-    return reduceMathExpression(node, precision);
-
-  return node;
-}
-
-function isEqual(left, right) {
-  return left.type === right.type && left.value === right.value;
-}
-
-function isValueType(type) {
-  switch (type) {
-    case 'LengthValue':
-    case 'AngleValue':
-    case 'TimeValue':
-    case 'FrequencyValue':
-    case 'ResolutionValue':
-    case 'EmValue':
-    case 'ExValue':
-    case 'ChValue':
-    case 'RemValue':
-    case 'VhValue':
-    case 'VwValue':
-    case 'VminValue':
-    case 'VmaxValue':
-    case 'PercentageValue':
-    case 'Value':
-      return true;
-  }
-  return false;
-}
-
-function convertMathExpression(node, precision) {
-  let nodes = convert(node.left, node.right, precision);
-  let left = reduce(nodes.left, precision);
-  let right = reduce(nodes.right, precision);
-
-  if (left.type === "MathExpression" && right.type === "MathExpression") {
-
-    if (((left.operator === '/' && right.operator === '*') ||
-      (left.operator === '-' && right.operator === '+')) ||
-      ((left.operator === '*' && right.operator === '/') ||
-      (left.operator === '+' && right.operator === '-'))) {
-
-      if (isEqual(left.right, right.right))
-        nodes = convert(left.left, right.left, precision);
-
-      else if (isEqual(left.right, right.left))
-        nodes = convert(left.left, right.right, precision);
-
-      left = reduce(nodes.left, precision);
-      right = reduce(nodes.right, precision);
-
-    }
-  }
-
-  node.left = left;
-  node.right = right;
-  return node;
-}
-
-function flip(operator) {
-  return operator === '+' ? '-' : '+';
-}
-
-function flipValue(node) {
-  if (isValueType(node.type))
-    node.value = -node.value;
-  else if (node.type === 'MathExpression') {
-    node.left = flipValue(node.left);
-    node.right = flipValue(node.right);
-  }
-  return node;
-}
-
-function reduceAddSubExpression(node, precision) {
-  const {left, right, operator: op} = node;
-
-  if (left.type === 'Function' || right.type === 'Function')
-    return node;

```

Comment:
Bad idea, because you can write `calc(var(--foo) - calc(var(--bar) + var(--baz)))`